### DISCUSSION TITLE
Limit number of selection in Checkbox group

### DISCUSSION THREAD

**Discussion by adeepika1991** [react-hook-form/react-hook-form#3545] [Q&A]
I have a functionality to implement which I don't know how to implement in react-hook-form. 

It is to limit no of user selection in checkbox eg, for a checkbox with 10 items and user can only select 3 and prevents selecting 4th item.

I could implement this in plain react but is there any register object property to specifically do this?

**Comment by Danziger** [comment]
Simple example, but I think what you are looking for is either [`watch`](https://react-hook-form.com/api/#watch) or [`useWatch`](https://react-hook-form.com/api/#useWatch):

```
export default function App() {
  const { watch, register } = useForm<{ toggles: boolean[] }>();
  const toggles = watch('toggles', []);
  const checkedCount = toggles.filter(Boolean).length;

  return (
    <form>
      { [...Array(10)].map((_, i) => (
        <label>
          <input
            ref={ register }
            type="checkbox"
            name={ `toggles[${ i }]` }
            disabled={ !toggles[i] && checkedCount >= 3 } />
          Checkbox { i + 1 }
        </label>
      )) }
    </form>
  );
}
```

Note this will re-render all your form every time the checkboxes change, so you might want to isolate that re-render as explained [here](https://react-hook-form.com/api/#useWatch).

Alternatively, you can allow selecting any number of checkboxes but prevent the form submission with validation. Using Yup on the previous example, you would add something like `yup.array(yup.boolean()).compact().min(1).max(3)`.

**Reply by bluebill1049** [reply]
Thanks for your support @Danziger 👍  Yes, you may need to watch how many checkboxes users have been checked.
